OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,375 @@
+                        hash = hash * _maxUsefulKey + code;
+                        if (Time.realtimeSinceStartup - _timeout[code] > _cooldown)
+                        {
+                            countdownTrigger = false;
+                        }
+                    }
+                }
+                if (!_comboRegistry.ContainsKey(hash))
+                {
+                    if (_lastPressed != null)
+                    {
+                        _lastPressed.SetPressed(false);
+                    }
+                    _lastPressed = null;
+                    return;
+                }
+                IModInputCombination combination = _comboRegistry[hash];
+                if (combination == _lastPressed || !countdownTrigger)
+                {
+                    if (_lastPressed != null && _lastPressed != combination)
+                    {
+                        _lastPressed.SetPressed(false);
+                    }
+                    _lastPressed = (ModInputCombination)combination;
+                    _lastPressed.SetPressed();
+                    if (keysCount > 1)
+                    {
+                        for (var i = 0; i < keysCount; i++)
+                        {
+                            _timeout[keys[i]] = Time.realtimeSinceStartup;
+                        }
+                    }
+                }
+            }
+        }
+
+        private bool IsPressed_Combo(IModInputCombination combination)
+        {
+            UpdateCombo();
+            return _lastPressed == combination;
+        }
+
+        private bool IsNewlyPressed_Combo(IModInputCombination combination, bool keep = false)
+        {
+            UpdateCombo();
+            return _lastPressed == combination && combination.IsFirst(keep);
+        }
+
+        private bool WasTapped_Combo(IModInputCombination combination)
+        {
+            UpdateCombo();
+            return combination != _lastPressed
+                && (Time.realtimeSinceStartup - combination.LastPressedMoment < _tapKeep)
+                && (combination.PressDuration < _tapDuration);
+        }
+
+        private bool IsNewlyReleased_Combo(IModInputCombination combination, bool keep = false)
+        {
+            UpdateCombo();
+            return _lastPressed != combination && combination.IsFirst(keep)
+                && (Time.realtimeSinceStartup - combination.LastPressedMoment < _tapKeep);
+        }
+
+        private bool IsPressed_Single(IModInputCombination combination)
+        {
+            if (Time.realtimeSinceStartup - _lastSingleUpdate > _updateCooldown)
+            {
+                _lastSingleUpdate = Time.realtimeSinceStartup;
+                var toRemove = new List<IModInputCombination>();
+                foreach (IModInputCombination combo in _singlePressed)
+                {
+                    if (!IsPressed(combo))
+                    {
+                        ((ModInputCombination)combo).SetPressed(false);
```

Comment:
So you're casting to ModInputCombination because SetPressed is not part of the interface... could you just add it?

Reply:
SetPressed if internal, can't add it to public interface

Reply:
and I can't inherit internal by public, so I would have to cast either way